src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -409,6 +410,39 @@ void VM_Version::initialize() {
     UsePopCountInstruction = true;
   }
 
+  if (UseBranchProtection == nullptr || strcmp(UseBranchProtection, "none") == 0) {
```

Comment:
My understanding is that `UseBranchProtection` requires JIT compiler integration. This is currently missing in Graal. The logic that decides if it should be enabled by default needs to take JVMCI into consideration, much like the call to `JVMCIGlobals::check_jvmci_supported_gc` determines if JVMCI supports the configured GC.

Thanks to @gilles-duboscq for alerting me to this change.

Reply:
Thanks for reporting this. @dougxc 

Yes. I encountered several jtreg failures with Graal, which is built with PAC-RET enabled JDK.

As I see it, the straightforward fix is to disable JVMCI if `_rop_protection` is parsed as "true" finally, since Graal doesn't supports PAC-RET currently. 
That is similar to the way `JVMCIGlobals::check_jvmci_supported_gc` does, i.e. if one GC except `G1/Serial/Parallel` is configured, we disable JVMCI.
WDYT

Reply:
The problem with `JVMCIGlobals::check_jvmci_supported_gc` is that it does not give the compiler a chance to specify whether it supports a given GC.
Instead, we want a JVMCI upcall like `jdk.vm.ci.hotspot.HotSpotJVMCIRuntime.isGCSupported(int)`.

Reply:
Thanks for your reply. But I'm afraid I'm still a bit confused.

1) flag `_rop_protection` is set inside function `VM_Version::initialize()`. The execution flow is `create_vm -> init_globals -> VM_Version_init -> VM_Version::initialize()`.
2) I suppose compilers (including JVMCI compiler) are initialized after `init_global()` function. See https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/threads.cpp#L685~L707. Hence I suppose we cannot call JVMCI methods before this site, right?

If so, I'm afraid it's a bit late for VM to ask the compiler whether it supports PAC-RET after the compiler initialization, and to disable PAC-RET if the JVMCI compiler doesn't support it, mainly because some stubs are generated already between `init_global()` and compiler initialization, and in these stubs, flag `_rop_protection` is used to generate PAC-related code, e.g., `SharedRuntime::generate_stubs() -> generate_deopt_blob() -> protect_return_address()`.

Please correct me if there is something I misunderstood. Thanks.

Reply:
Ah ok, I did not pay attention to the VM lifecycle constraints in setting this flag. We could do the JVMCI check in `CompileBroker::compilation_init_phase2` but that would mean forcing eager initialization of JVMCI which, in turn, would impact VM startup.
For now, we will just add a check into Graal that `UseBranchProtection` is not set, exiting the VM if it is. At the same time, we will work on adding support for this.

Reply:
>For now, we will just add a check into Graal that UseBranchProtection is not set, exiting the VM if it is

that sounds a better solution to me instead of:

> disable JVMCI if _rop_protection is parsed as "true"